graphql/src/store/prefetch.rs

```diff
@@ -658,17 +602,73 @@ fn execute_selection_set<'a>(
     }
 }
 
+#[derive(Default)]
+struct CollectedResponseKey<'a> {
+    interface_cond: Option<&'a s::InterfaceType>,
+    interface_fields: Vec<&'a q::Field>,
+    concrete_types: IndexMap<ObjectCondition<'a>, Vec<&'a q::Field>>,
+}
+
+impl<'a> CollectedResponseKey<'a> {
+    fn collect_field(&mut self, type_condition: ObjectOrInterface<'a>, field: &'a q::Field) {
+        match type_condition {
+            ObjectOrInterface::Interface(i) => {
+                self.interface_cond = Some(i);
```

Comment:
Why is it ok to clobber a potentially already existing `Some(_)` here?

Reply:
I've added an assert that if this assigns twice, it will assign the same value. It's not a very obvious invariant but `collect_fields` will never call this for two different interface types.